### PR TITLE
Add documentation of just_once

### PR DESCRIPTION
**Pull request by prescod**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+### Singletons with the "just_once" feature
+
+Snowfakery scales up to larger data volumes
+by evaluating your recipe over and over again.
+Each one is called an `iteration`.
+
+Some objects are meant to be produced only once, regardless
+of how many times the recipe executes. The programming
+language term for thhis is a "singleton". For example
+an accounting system might generate a datase that has
```
Comment (prescod) [inline docs/index.md:1027]:
```suggestion
an accounting system might generate a dataset that has
```

File: docs/index.md [hunk 2]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+
+### Singletons with the "just_once" feature
+
+Snowfakery scales up to larger data volumes
+by evaluating your recipe over and over again.
+Each one is called an `iteration`.
+
+Some objects are meant to be produced only once, regardless
+of how many times the recipe executes. The programming
+language term for thhis is a "singleton". For example
```
Comment (prescod) [inline docs/index.md:1026]:
```suggestion
language term for this is a "singleton". For example
```

File: docs/index.md [hunk 3]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+
+Snowfakery scales up to larger data volumes
+by evaluating your recipe over and over again.
+Each one is called an `iteration`.
+
+Some objects are meant to be produced only once, regardless
+of how many times the recipe executes. The programming
+language term for this is a "singleton". For example
+an accounting system might generate a dataset that has
+exactly 3 Ledger objects, Checking and Savings. You could have
```
Comment (Br4nd0R) [inline docs/index.md:1028]: Do you mean 2 here? Or is it 3 Ledger object + 1 checking and 1 savings?
```suggestion
exactly 2 Ledger objects, Checking and Savings. You could have
```

File: docs/index.md [hunk 4]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+dozens of Entries per Ledger or maybe billions. But it
+might want to always generate exactly 3 Ledgers.
```
Comment (Br4nd0R) [inline docs/index.md:1030]: same question here.

File: docs/index.md [hunk 5]
```diff
@@ -1452,6 +1506,8 @@ $ cci flow run test_everything
 - Object Template: These represent instructions on how to create a row, or multiple rows in a database. Each row represents a real-world Object.
 - Rows: Rows (often also called “records”) in a database are a unit of related information. For example in Salesforce (which includes a database) a “Contact” has a first name, last name, phone number, etc. Each Contact is a row. “Contact” is the type of each of those rows. Rows represent real-world Objects. See “Objects” above for more information.
 - Recipe: A Snowfakery YAML file instructing Snowfakery on what to generate.
+- Iteration: Snowfakery recipes can be "scaled up" to generate more data by specifying command line, API or CumulusCI options. The recipe is scaled up by executing over and over again. These executions are called iterations.
+- Singleton: A singleton is an Object Template that generates a single row now how many times the recipe is iterated over.
```
Comment (Br4nd0R) [inline docs/index.md:1510]:
```suggestion
- Singleton: A singleton is an Object Template that generates a single row regardless of how many times the recipe is iterated over.
```

**Review by Br4nd0R — CHANGES_REQUESTED** [review]
some minor stuff. Looks great otherwise!

File: docs/index.md [hunk 6]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+    Ledger: Savings
+    ...
+```
+
+Now if you execute this from the Snowfakery command line like this:
+
+```s
+$ snowfakery accounting.yml --target-number 10_000 Entry
+...
+```
```
Comment (Br4nd0R) [inline docs/index.md:1065]:
```suggestion
```
$ snowfakery accounting.yml --target-number 10_000 Entry
...
```
```

File: docs/index.md [hunk 7]
```diff
@@ -1014,6 +1014,60 @@ for all of the CSV files.
...
+- object: Entry
+  count: 1000
+  fields:
+    Ledger: Savings
+    ...
+```
+
+Now if you execute this from the Snowfakery command line like this:
+
+```s
```
Comment (Br4nd0R) [inline docs/index.md:1062]: I don't think the 's' is needed.

File: docs/index.md (hunk 4 above)
Comment (prescod) [inline docs/index.md:1030]:
```suggestion
might want to always generate exactly 2 Ledgers.
```

File: docs/index.md (hunk 7 above)
Comment (prescod) [reply docs/index.md:1062]: It seems to format the text as a shell script. I guess I could have called it "sh" to be clearer.

Comment (prescod) [reply docs/index.md:1062]: Experimentally "s" works best.

    ```s 

```s 
$ foo bar baz 
xyzzy
```

```bash
$ foo bar baz 
xyzzy
```

```sh
$ foo bar baz 
xyzzy
```

works.

Comment (Br4nd0R) [reply docs/index.md:1062]: TIL

**Review by Br4nd0R — APPROVED** [review]

**Pull request merged**